src/firestarter/Environment/X86/Payload/AVX512Payload.cpp

```diff
@@ -388,4 +448,65 @@ void AVX512Payload::init(double* MemoryAddr, uint64_t BufferSize) const {
   X86Payload::initMemory(MemoryAddr, BufferSize, 0.27948995982e-4, 0.27948995982e-4);
 }
 
-} // namespace firestarter::environment::x86::payload
\ No newline at end of file
+void AVX512Payload::create_AMX_config(__tilecfg* tileinfo) {
+  // Create tile_cfg, fill it and return
+  int i;
+  tileinfo->palette_id = 1;
+  tileinfo->start_row = 0;
+
+  for (i = 0; i < 8; ++i) {
+    tileinfo->colsb[i] = MAX_COLS;
+    tileinfo->rows[i] = MAX_ROWS;
+  }
+
+  _tile_loadconfig(tileinfo);
+}
+
+void AVX512Payload::request_permission() {
+
+  long rc;
+  unsigned long bitmask;
+  rc = syscall(SYS_arch_prctl, ARCH_REQ_XCOMP_PERM, XFEATURE_XTILEDATA);
+
+  if (rc) {
+    workerLog::error() << "XTILE_DATA request failed: " << rc;
+  }
+
+  rc = syscall(SYS_arch_prctl, ARCH_GET_XCOMP_PERM, &bitmask);
+  if (rc) {
+    workerLog::error() << "prctl(ARCH_GET_XCOMP_PERM) error: " << rc;
+  }
+  if (bitmask & XFEATURE_MASK_XTILE) {
+    workerLog::trace() << "ARCH_REQ_XCOMP_PERM XTILE_DATA successful.";
+  } else {
+    workerLog::error() << "[ERROR] ARCH_REQ_XCOMP_PERM XTILE_DATA unsuccessful!";
+  }
+}
+
+void AVX512Payload::init_buffer_rand(uintptr_t src1, uintptr_t src2) {
```

Comment:
Pointers to the function should be `__bfloat16*` This may however not be supported by all compilers. We might need to initialize this memory differently.